httpx/_client.py

```diff
@@ -66,6 +74,7 @@ def __init__(
         cookies: CookieTypes = None,
         timeout: TimeoutTypes = DEFAULT_TIMEOUT_CONFIG,
         max_redirects: int = DEFAULT_MAX_REDIRECTS,
+        connect_retries: int = DEFAULT_CONNECT_RETRIES,
```

Comment:
Perhaps `retries: int = DEFAULT_RETRIES` is okay.
That gives us more scope to later keep the same API, but extend it to also support more granular retry behaviours? (But with a plain `retries=<int>` always just using basic connect retry behaviour)